Description: Upstream Changeset 2608
  make sure pdns_server --version works even if guardian or
  daemon are enabled.
Origin: http://wiki.powerdns.com/trac/changeset/2608/

--- a/pdns/receiver.cc
+++ b/pdns/receiver.cc
@@ -417,6 +417,15 @@
       
     ::arg().laxParse(argc,argv); // do a lax parse
     
+    if(::arg().mustDo("version")) {
+      cerr<<"Version: "VERSION", compiled on "<<__DATE__", "__TIME__;
+#ifdef __GNUC__ 
+      cerr<<" with gcc version "<<__VERSION__;
+#endif
+      cout<<endl;
+      exit(99);
+    }
+
     if(::arg()["config-name"]!="") 
       s_programname+="-"+::arg()["config-name"];
     
@@ -455,14 +464,6 @@
       cerr<<"Um, we did get here!"<<endl;
     }
 
-    if(::arg().mustDo("version")) {
-      cerr<<"Version: "VERSION", compiled on "<<__DATE__", "__TIME__;
-#ifdef __GNUC__ 
-      cerr<<" with gcc version "<<__VERSION__;
-#endif
-      cout<<endl;
-      exit(99);
-    }
     
     // we really need to do work - either standalone or as an instance
 
